tests/httpInterface.patchById.special.test.js

```diff
@@ -677,6 +677,221 @@ tap.test('HTTP PATCH /<id> - nested object', async t => {
+        t.strictSame(metadata.exampleArrayOfArray, [
+          ['item1'],
+          ['new-item'],
+        ])
+
+        const docOnDb = await collection.findOne({ _id: DOC_TEST._id })
+        t.strictSame(docOnDb.metadata.exampleArrayOfArray, [
+          ['item1'],
+          ['new-item'],
+        ])
+
+        t.end()
+      })
+
+      t.test('push to child array', async t => {
+        const { fastify, collection } = await setUpTest(t, [DOC_TESTING_ARRAY])
+        const UPDATE_COMMAND = {
+          $addToSet: {
+            'metadata.exampleArrayOfArray.0': 'new-item',
+          },
+        }
+        const response = await fastify.inject({
+          method: 'PATCH',
+          url: `${prefix}/${DOC_TESTING_ARRAY._id.toHexString()}`,
+          payload: UPDATE_COMMAND,
+          headers: {
+            userId: newUpdaterId,
+          },
+        })
+        t.equal(response.statusCode, 200)
+
+        const { metadata } = JSON.parse(response.payload)
+        t.strictSame(metadata.exampleArrayOfArray, [
+          ['item1', 'new-item'],
+        ])
+
+        const docOnDb = await collection.findOne({ _id: DOC_TEST._id })
+        t.strictSame(docOnDb.metadata.exampleArrayOfArray, [
+          ['item1', 'new-item'],
+        ])
+
+        t.end()
+      })
+
+      t.end()
+    })
+
+    t.test('fails if missing to push a required field of an object', async t => {
+      const { fastify } = await setUpTest(t)
+      const UPDATE_COMMAND = {
+        $addToSet: {
+          attachments: {
+            other: 'foo',
+            // name is required
+          },
+        },
+      }
+      const response = await fastify.inject({
+        method: 'PATCH',
+        url: `${prefix}/${DOC_TEST._id.toHexString()}`,
+        payload: UPDATE_COMMAND,
+        headers: {
+          userId: newUpdaterId,
+        },
+      })
+      t.equal(response.statusCode, 400)
+      t.strictSame(JSON.parse(response.payload), {
+        statusCode: 400,
+        error: 'Bad Request',
+        message: "body must have required property 'name'",
+      })
+      t.end()
+    })
+
+    t.test('fails if push is not compliant to additionalProperties of the field', async t => {
+      const { fastify } = await setUpTest(t)
+      const UPDATE_COMMAND = {
+        $addToSet: {
+          attachments: {
+            name: 'foo',
+            unknownField: 'bar',
+            // additionalProperties is false
+          },
+        },
+      }
+      const response = await fastify.inject({
+        method: 'PATCH',
+        url: `${prefix}/${DOC_TEST._id.toHexString()}`,
+        payload: UPDATE_COMMAND,
+        headers: {
+          userId: newUpdaterId,
+        },
+      })
+      t.equal(response.statusCode, 400)
+      t.strictSame(JSON.parse(response.payload), {
+        statusCode: 400,
+        error: 'Bad Request',
+        message: 'body must NOT have additional properties',
+      })
+      t.end()
+    })
+    t.end()
+  })
```

Comment:
Would it be possible to add a test that verifies the PATCH operation fails in case the field interested by `$addToSet` operator is not an array?

Reply:
Yes sure, it's definitely to be done.

Reply:
I am convinced that the error message should be more specific in this case. But it would require more substantial changes to how inputs are validated, so I think it may be the subject of another issue.